docs/operations/debug.md

```diff
@@ -91,4 +116,107 @@ $> kubectl logs -n <KubeVirt Install Namespace> virt-handler-2m86x | head -n8
 ```
 
 Obviously, for both examples above, `<KubeVirt Install Namespace>` needs to be replaced with the actual namespace
-KubeVirt is installed in.
\ No newline at end of file
+KubeVirt is installed in.
+
+
+## KubeVirt PProf Profiler
+Using the `cluster-profiler` client tool, a developer can get the PProf profiling data for every component in the Kubevirt Control plane. Here is a user guide:
+
+### Compile `cluster-profiler`
+Build from source code
+```
+git clone https://github.com/kubevirt/kubevirt.git
+cd kubevirt/tools/cluster-profiler
+go build
+```
+
+### Enable the feature gate
+Add `ClusterProfiler` in KubeVirt config
+```
+cat << END > enable-feature-gate.yaml
+---
+apiVersion: kubevirt.io/v1
+kind: KubeVirt
+metadata:
+  name: kubevirt
+  namespace: kubevirt
+spec:
+  configuration:
+    developerConfiguration:
+      featureGates:
+        - ClusterProfiler
+END
+
+kubectl apply -f enable-feature-gate.yaml
+```
+
+### Do the profiling
+Start CPU profiling
+```
+$ cluster-profiler --cmd start
```

Comment:
It improves readability to include a blank line between commands and their output.